### PR TITLE
Refactored WP_Debug_Data::debug_data() Part 2

### PR DESCRIPTION
**Pull request by apermo**
Changed: Moving $info['wp-constants'] to own function

---

Follow up to https://github.com/WordPress/wordpress-develop/pull/7027 as discussed with @dmsnell and @costdev 

Summary of the coming steps:

1. [wp-filesystem](https://github.com/WordPress/wordpress-develop/pull/7065)
2. **wp-constants** (we are here)
3. wp-database
4. wp-server
5. wp-media
6. wp-plugins-active & wp-plugins-inactive
7. wp-mu-plugins
8. wp-themes-inactive
9. wp-parent-theme
10. wp-active-theme
11. wp-dropins
12. wp-paths-sizes
13. wp-core

Refactoring WP_Debug_Data::debug_data(); into smaller functions for

* better maintainability
* reduced complexity
* preparation for future unit tests
* improved extensibility (see trac for details)
* added php7.2 style type hints
* fixed an error with `$parent_theme_auto_update_string` showing the value from `$auto_updates_string` (This will be handled only in Step 9)

Trac ticket: https://core.trac.wordpress.org/ticket/61648#ticket

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-admin/includes/class-wp-debug-data.php [hunk 1]
```diff
@@ -1436,6 +1296,157 @@ public static function debug_data() {
 		return $info;
 	}
 
+	/**
+	 * Gets the WP constants section of the debug data.
```
Comment (costdev) [inline src/wp-admin/includes/class-wp-debug-data.php]:
```suggestion
	 * Gets the WordPress constants section of the debug data.
```
Nit: Let's use the full application name in the docblock.

Comment (apermo) [reply src/wp-admin/includes/class-wp-debug-data.php]: Absolutely. Applied it.

**Review by dmsnell — APPROVED** [review]
After reviewing for dangling references I found none. This change seems sound. Thanks again @apermo - let's move to part 3

**Comment by dmsnell** [comment]
Merged in [[58855]](https://core.trac.wordpress.org/changeset/58855)
https://github.com/wordpress/wordpress-develop/commit/8aca1cf6bc78a4e9a32dd47e37e74ea8e24d0a62

**Pull request closed without merging**